src/core/symbol/SymbolTransaction.ts

```diff
@@ -0,0 +1,357 @@
+    }
+
+    /**
+     * It converts an Body Builder (e.g: TransferTransactionBodyBuilder) to an Embedded Builder (e.g: EmbeddedTransferTransactionBuilder)to be used in an aggregate transaction
+     * @param bodyBuilder - the body builder
+     * @param signerPublicKey - the signer public key.
+     */
+    public toEmbedded(bodyBuilder: Serializer, signerPublicKey: Key): EmbeddedTransactionBuilder {
+        return SymbolTransactionUtils.createEmbeddedFromBodyBuilder({
+            network: this.network.identifier,
+            signerPublicKey: new KeyDto(signerPublicKey.key),
+            bodyBuilder: bodyBuilder,
+        });
+    }
+
+    /**
+     * Helper method that serializes an unresolved address to be used in Catbuffer builders.
+     * @param unresolvedAddressId - the unresolved address to serialize.
+     */
+    public toUnresolvedAddress(unresolvedAddressId: SymbolUnresolvedAddress): Uint8Array {
+        if (typeof unresolvedAddressId == 'bigint') {
+            return SymbolIdGenerator.encodeUnresolvedAddress(this.network.identifier, unresolvedAddressId);
+        } else {
+            return unresolvedAddressId.getAddressBytes();
+        }
+    }
+
+    /**
+     * Generic method to create transaction from a top level (simple or aggregate) builder.
+     * @param builder - the builder
+     */
+    public createFromBuilder<T extends TransactionBuilder>(builder: T): SymbolTransaction<T> {
+        return SymbolTransaction.createFromBuilder(this.network, builder);
+    }
+
+    /**
+     * Generic method to create transaction from a top level (simple or aggregate) payload.
+     *
+     * @param payload - the serialized payload.
+     */
+    public createFromPayload(payload: Uint8Array): SymbolTransaction {
+        return SymbolTransaction.createFromPayload(this.network, payload);
+    }
+
+    /**
+     * Hides how transactions hash is generated from the provided embedded transactions. Dev doesn't need to calculate at front.
+     * @param embeddedBuilders - the builders of the embedded transactions.
+     */
+    private calculateAggregateTransactionsHashFromPayloads(embeddedBuilders: EmbeddedTransactionBuilder[]): Uint8Array {
+        return SymbolTransactionUtils.calculateAggregateTransactionsHashFromPayloads(embeddedBuilders.map((b) => b.serialize()));
+    }
+}
+
+/**
+ * The mutable generic symbol transaction that wraps the signing over catbuffer builders.
+ */
+export class SymbolTransaction<T extends TransactionBuilder = TransactionBuilder> {
+    /**
+     * The network generation hash used for resolve the transaction hash and the signing data of the transaction.
+     */
+    private readonly generationHash: Uint8Array;
+
+    /**
+     * The serialized payload of a top level simpler or aggregate transaction. Always in sync with the catbuffer builder.
+     *
+     * This field changes when signing and cosigning the transaction.
+     *
+     */
+    private _payload: Uint8Array;
```

Comment:
are we using _ in ts?

Better to move the private accessor into the constructor and remove these?

Reply:
The issue is that I want payload and builder to be read-only accessible form the outside word, the user should not be able change it. I don't like the _ either. I can rename it to internalPayload/internalBuilder or use a SymbolTransaction interface.